### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by jednano**
Based on #16 

See line comments in e1fd7a8 for patterns used.

### CODE REVIEW

File: src/editorConfigMain.ts [hunk 1]
```diff
@@ -26,247 +37,271 @@ interface IEditorConfigProvider {
...
-
-            applyEditorConfigToTextEditor(window.activeTextEditor, this);
-        });
-    }
+	private _documentToConfigMap: { [uri: string]: editorconfig.knownProps };
+	private _disposable: Disposable;
+
+	constructor() {
+
+		let subscriptions: Disposable[] = []
```
Comment (SamVerschueren) [inline src/editorConfigMain.ts]: This can be `const` as the reference to `subscriptions` never changes.

File: src/editorConfigMain.ts [hunk 2]
```diff
@@ -26,247 +37,271 @@ interface IEditorConfigProvider {
...
+		);
+	}
+
+	private _onDidOpenDocument(document: TextDocument): void {
+		if (document.isUntitled) {
+			// Does not have a fs path
+			return;
+		}
+
+		let path = document.fileName;
```
Comment (SamVerschueren) [inline src/editorConfigMain.ts]: `const`

File: src/editorConfigMain.ts [hunk 3]
```diff
@@ -26,247 +37,271 @@ interface IEditorConfigProvider {
...
+	let doc = textEditor.document;
+	let editorconfig = provider.getSettingsForDocument(doc);
+
+	if (!editorconfig) {
+		// no configuration found for this file
+		return;
+	}
+
+	let { insertSpaces, tabSize } = textEditor.options;
+	let newOptions = Utils.fromEditorConfig(
```
Comment (SamVerschueren) [inline src/editorConfigMain.ts]: `const` here as well as 114, 115, 122

File: src/editorConfigMain.ts [hunk 4]
```diff
@@ -26,247 +37,271 @@ interface IEditorConfigProvider {
...
 
 function applyOnSaveTransformations(
-    textDocument:TextDocument,
-    provider:IEditorConfigProvider): void {
+	textDocument:TextDocument,
+	provider:IEditorConfigProvider
+): void {
 
-    let editorconfig = provider.getSettingsForDocument(textDocument);
+	let editorconfig = provider.getSettingsForDocument(textDocument);
```
Comment (SamVerschueren) [inline src/editorConfigMain.ts]: `const`

**Comment by SamVerschueren** [comment]
I'd suggest using `const` as much as possible. Added some notes but it's just too much to do them all :).

**Comment by egamma** [comment]
> I'd suggest using `const`  as much as possible. Added some notes but it's just too much to do them all :).

May be the project should enable tslint. There is a lint rule for this check in the TypeScript repository.
[preferConstRule](https://github.com/Microsoft/TypeScript/blob/master/scripts/tslint/preferConstRule.ts)

**Comment by SamVerschueren** [comment]
@egamma good idea :+1:

**Comment by jednano** [comment]
@SamVerschueren I replaced all the lets with const where I could in fb40989, but I think tslint is out of the scope of this PR. Feel free to open an issue concerning tslint.

**Comment by chrisdias** [comment]
closing this PR as all work has moved to https://github.com/editorconfig/editorconfig-vscode

**Pull request closed without merging**